genesyscloud/resource_genesyscloud_journey_segment.go

```diff
@@ -0,0 +1,623 @@
+	"github.com/mypurecloud/terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+)
+
+var (
+	journeySegmentSchema = map[string]*schema.Schema{
+		"is_active": {
+			Description: "Whether or not the segment is active.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+		},
+		"display_name": {
+			Description: "The display name of the segment.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"description": {
+			Description: "A description of the segment.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"color": {
+			Description: "The hexadecimal color value of the segment.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ValidateFunc: validation.StringMatch(func() *regexp.Regexp {
+				r, _ := regexp.Compile("^#[a-fA-F\\d]{6}$")
+				return r
+			}(), ""),
+		},
+		"scope": {
+			Description:  "The target entity that a segment applies to.Valid values: Session, Customer.",
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true, // scope can be only set during creation
+			ValidateFunc: validation.StringInSlice([]string{"Session", "Customer"}, false),
+		},
+		"should_display_to_agent": {
+			Description: "Whether or not the segment should be displayed to agent/supervisor users.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			// Customer scope only supports false for this value
+		},
+		"context": {
+			Description: "The context of the segment.",
+			Type:        schema.TypeSet,
+			Optional:    true,
+			MaxItems:    1,
+			Elem:        contextResource,
+		},
+		"journey": {
+			Description: "The pattern of rules defining the segment.",
+			Type:        schema.TypeSet,
+			Optional:    true,
+			MaxItems:    1,
+			Elem:        journeyResource,
+		},
+		"external_segment": {
+			Description: "Details of an entity corresponding to this segment in an external system.",
+			Type:        schema.TypeSet,
+			Optional:    true,
+			MaxItems:    1,
+			Elem:        externalSegmentResource, // can only be used with Customer scope
+		},
+		"assignment_expiration_days": {
+			Description: "Time, in days, from when the segment is assigned until it is automatically unassigned.",
+			Type:        schema.TypeInt,
+			Optional:    true,
+		},
+	}
+
+	contextResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"patterns": {
+				Description: "A list of one or more patterns to match.",
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        contextPatternResource,
+			},
+		},
+	}
+
+	journeyResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"patterns": {
+				Description: "A list of zero or more patterns to match.",
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        journeyPatternResource,
+			},
+		},
+	}
+
+	externalSegmentResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
```

Comment:
What is this the ID of? If it's another resource that we haven't already created in the provider then we'll have to create it also

Reply:
Let me know if you need to write more resources. I'm working on a code generator that will create a lot of the boiler plate for you.